core/domain/change_domain.py

```diff
@@ -226,6 +236,7 @@ def to_dict(self):
     def __getattr__(self, name: str) -> str:
         # AttributeError needs to be thrown in order to make
         # instances of this class picklable.
+        assert (isinstance(name, str) and isinstance(getattr, str))
```

Comment:
Because, the original code already had type-annotation of `getattr` as `str`, but it Mypy was throwing the following error

![image](https://user-images.githubusercontent.com/65958420/148418205-60fc850f-98e7-49b1-bb06-d67ade3989cb.png)

That's why I used this statement, to get rid of this error, as suggested by @sajalasati.

Reply:
But `getattr` is of `Callable` type and it is a builtin function, it doesn't make sense to check it against `str` (The first part of this is fine.

Reply:
But it is returning `Any`, by default. I have used this, to use `str` as Mypy return type for the function. What do you suggest?

Reply:
But you're not calling `getattr` anywhere so why do you care about its return value?

Also this will be always false:
![image](https://user-images.githubusercontent.com/10142938/148927079-f130caac-84cc-4d00-926b-bc78dfc83a0d.png)

Reply:
So, can we use `# type: ignore [no-untyped-def]` in this case? As, we are not caring about its return value and Mypy would throw error if we don't type annotate it, and we cannot use `Any` as well.

Reply:
What is returned when you do `reveal_type(self.__dict__[name])` and run the mypy checks?

Reply:
This is getting returned
![image](https://user-images.githubusercontent.com/65958420/149807360-83c8c833-437d-4bba-96b5-ba08757078c8.png)

Reply:
Yeah I think lets return `Any` from this function but lets add a proper explanatory comment for it.

Reply:
Done and pushed!